### PR TITLE
Add endpoints for new VAC resources API

### PR DESCRIPTION
**Pull request by JoMas971**
Do not merge this PR yet. We need to re-wire everything console side first.

### CODE REVIEW

File: src/sdk/model/vac/__json__/ba-backup-repository.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  id: string;
+  name: string;
+  repository_uid: string;
+  server_name: string;
+  company_name: string;
+  location_name: string;
+  capacity: number;
+  capacity_units: string;
+  free_space: number;
+  free_space_units: string;
```
Comment (cfsnyder) [inline src/sdk/model/vac/__json__/ba-backup-repository.ts]: this field does not exist on the server response

File: src/sdk/model/vac/__json__/ba-backup-repository.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export interface BaBackupRepositoryJson {
+  id: string;
+  name: string;
+  repository_uid: string;
+  server_name: string;
+  company_name: string;
+  location_name: string;
+  capacity: number;
+  capacity_units: string;
```
Comment (cfsnyder) [inline src/sdk/model/vac/__json__/ba-backup-repository.ts]: this field does not exist on the server response

File: src/sdk/model/vac/__json__/ba-backup-repository.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  free_space: number;
+  free_space_units: string;
+  backup_size: number;
+  backup_size_units: string;
```
Comment (cfsnyder) [inline src/sdk/model/vac/__json__/ba-backup-repository.ts]: doesn't exist

File: src/sdk/model/vac/__json__/ba-backup-resource.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { BaBackupRepositoryJson } from './ba-backup-repository';
+import { BaWanAcceleratorJson } from './ba-wan-accelerator';
+
+export interface BaBackupResourceJson {
+  id: string;
+  cloud_repository_name: string;
+  storage_quota: number;
+  storage_quota_units: string;
```
Comment (cfsnyder) [inline src/sdk/model/vac/__json__/ba-backup-resource.ts]: doesn't exist

File: src/sdk/model/vac/__json__/ba-backup-resource.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  vms_quota: number;
+  workstations_quota: number;
+  traffic_quota: number;
+  traffic_quota_units: string;
+  wan_acceleration_enabled: boolean;
+  used_storage_quota: number;
+  used_storage_quota_units: string;
```
Comment (cfsnyder) [inline src/sdk/model/vac/__json__/ba-backup-resource.ts]: doesn't exist

File: src/sdk/model/vac/__json__/ba-backup-resource.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  used_traffic_quota: number;
+  used_traffic_quota_units: string;
```
Comment (cfsnyder) [inline src/sdk/model/vac/__json__/ba-backup-resource.ts]: doesn't exist

File: src/sdk/model/vac/__json__/ba-backup-resource.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+export interface BaBackupResourceJson {
+  id: string;
+  cloud_repository_name: string;
+  storage_quota: number;
+  storage_quota_units: string;
+  vms_quota: number;
+  workstations_quota: number;
+  traffic_quota: number;
+  traffic_quota_units: string;
```
Comment (cfsnyder) [inline src/sdk/model/vac/__json__/ba-backup-resource.ts]: doesn't exist

File: src/sdk/model/vac/ba-company-backup-history.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+   */
+  get lastResult(): string {
+    return this._json.last_result;
+  }
+
+  /**
+   * Get last active.
+   * @returns {number}
+   */
+  get lastActive(): number {
```
Comment (cfsnyder) [inline src/sdk/model/vac/ba-company-backup-history.ts]: should be returning a Date here

File: src/sdk/model/vac/ba-company.ts [hunk 9]
```diff
@@ -0,0 +1,512 @@
...
+   */
+  get backupResources(): Array<BaBackupResource> {
+    return this._json.backup_resources.map(it => new BaBackupResource(it));
+  }
+
+  /**
+   * Get last active.
+   * @returns {number}
+   */
+  get lastActive(): number {
```
Comment (cfsnyder) [inline src/sdk/model/vac/ba-company.ts]: should be returning a Date

File: src/sdk/model/vac/ba-company.ts [hunk 10]
```diff
@@ -0,0 +1,512 @@
...
+   */
+  get expirationEnabled(): boolean {
+    return this._json.expiration_enabled;
+  }
+
+  /**
+   * Get expiration date.
+   * @returns {number}
+   */
+  get expirationDate(): number {
```
Comment (cfsnyder) [inline src/sdk/model/vac/ba-company.ts]: Date

File: src/sdk/model/vac/ba-company.ts [hunk 11]
```diff
@@ -0,0 +1,512 @@
...
+  async refresh(): Promise<BaCompany> {
+    return Iland.getHttp().get(`/vac-companies/${ this.uuid }`).then((response) => {
+      this._json = response.data as BaCompanyJson;
+      return this;
+    });
+  }
+
+  /**
+   * Gets the storage usage for the Ba Company
+   * @param {number} start
```
Comment (cfsnyder) [inline src/sdk/model/vac/ba-company.ts]: fill in the param docs

File: src/sdk/model/vac/ba-company.ts [hunk 12]
```diff
@@ -0,0 +1,512 @@
...
+    }
+    return Iland.getHttp().get(`/vac-companies/${ this.uuid }/storage-usage`, requestParams).then((response) => {
+      const samples = response.data.data as Array<VacPerfSampleJson>;
+      return samples.map(it => new VacPerfSample(it));
+    });
+  }
+
+  /**
+   * Gets the backup history for the Ba Company
+   * @param {number} offset
```
Comment (cfsnyder) [inline src/sdk/model/vac/ba-company.ts]: fill in param docs

File: src/sdk/model/vac/ba-company.ts [hunk 13]
```diff
@@ -0,0 +1,512 @@
...
+   * @param {number} limit
+   */
+  async getBackupHistory(offset: number, limit: number): Promise<Array<BaCompanyBackupHistory>> {
```
Comment (cfsnyder) [inline src/sdk/model/vac/ba-company.ts]: offset and limit should be optional, right?

File: src/sdk/model/vac/ba-job.ts [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+   */
+  get type(): string {
+    return this._json.type;
+  }
+
+  /**
+   * Get last run.
+   * @returns {number}
+   */
+  get lastRun(): number {
```
Comment (cfsnyder) [inline src/sdk/model/vac/ba-job.ts]: Date

File: src/sdk/model/vac/ba-job.ts [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+    return this._json.last_run;
+  }
+
+  /**
+   * Get end time.
+   * @returns {number}
+   */
+  get endTime(): number {
```
Comment (cfsnyder) [inline src/sdk/model/vac/ba-job.ts]: Date

File: src/sdk/model/vac/vac-perf-sample.ts [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+   */
+  get quota(): number {
+    return this._json.quota;
+  }
+
+  /**
+   * Get time.
+   * @returns {number}
+   */
+  get time(): number {
```
Comment (cfsnyder) [inline src/sdk/model/vac/vac-perf-sample.ts]: Date

File: src/sdk/model/vac/__json__/ba-backup-repository.ts (hunk 1 above)
Comment (JoMas971) [reply src/sdk/model/vac/__json__/ba-backup-repository.ts]: Hum, it has been updated lately. I had the old version of it. Thank you for reporting it ! I'll update all of that.

File: src/sdk/model/vac/ba-company.ts (hunk 13 above)
Comment (JoMas971) [reply src/sdk/model/vac/ba-company.ts]: indeed, you're right. Will make them optional then.

File: src/sdk/model/vac/ba-job.ts (hunk 14 above)
Comment (JoMas971) [reply src/sdk/model/vac/ba-job.ts]: i forgot that we wanted to return Date object there... API side it is a timestamp so i didn't pay attention to it. Thank you for reporting, i'll update that.

**Comment by JoMas971** [comment]
@cfsnyder I've addressed all your comments. Should be good now.

File: src/sdk/model/vac/ba-job.ts (hunk 14 above)
Comment (cfsnyder) [reply src/sdk/model/vac/ba-job.ts]: it doesn't look like you did update it?

**Comment by JoMas971** [comment]
@cfsnyder I've updated the PR. It's ready for a new review ;-) Thanks

**Pull request merged**